Clean up RenderBox overflow methods

fantasai opened the pull request:
#### dd50ed74020270b2d08e8aff0f06811216df2bdd
<pre>
Clean up RenderBox overflow methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=300835">https://bugs.webkit.org/show_bug.cgi?id=300835</a>
<a href="https://rdar.apple.com/162722242">rdar://162722242</a>

Reviewed by Alan Baradlay.

The various computeOverflow/addOverflow methods are kinda tangled up. Clean it up.
- Shifts method declarations to the correct superclass/subclass level
  (adjusting names as necessary), and in the right section of the .h file.
- Simplifies parameters and sets us up for additional options in the future.
- Encapsulates memory management of m_overflow.

* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::computeOverflow):
(WebCore::RenderBlock::addOverflowFromOutOfFlowBoxes):
(WebCore::RenderBlock::addVisualOverflowFromTheme):
(WebCore::RenderBlock::simplifiedLayout):
(WebCore::RenderBlock::addOverflowFromChildren): Deleted.
(WebCore::RenderBlock::addOverflowFromBlockChildren): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
(WebCore::RenderBlock::computeOverflow):
(WebCore::RenderBlock::addOverflowFromInlineChildren): Deleted.
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::computeOverflow):
(WebCore::RenderBlockFlow::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderBlockFlow.h:
(WebCore::RenderBlockFlow::computeOverflow):
(WebCore::RenderBlockFlow::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::addOverflowFromInFlowChildren):
(WebCore::RenderBox::addOverflowFromContainedBox):
(WebCore::RenderBox::addOverflowWithRendererOffset):
(WebCore::RenderBox::addVisualOverflow):
(WebCore::RenderBox::percentageLogicalHeightIsResolvable const):
(WebCore::RenderBox::hasUnsplittableScrollingOverflow const):
(WebCore::RenderBox::isUnsplittableForPagination const):
(WebCore::RenderBox::addOverflowFromInFlowChildOrAbsolutePositionedDescendant): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::contentOverflowRect const):
(WebCore::RenderGrid::computeOverflow): Deleted.
* Source/WebCore/rendering/RenderGrid.h:
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::addOverflowFromInFlowChildren):
(WebCore::RenderMultiColumnSet::addOverflowFromChildren): Deleted.
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::addOverflowFromInFlowChildren):
(WebCore::RenderTable::addOverflowFromChildren): Deleted.
* Source/WebCore/rendering/RenderTable.h:
(WebCore::RenderTable::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderTableSection.cpp:
(WebCore::RenderTableSection::computeOverflowFromCells):
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
(WebCore::RenderSVGBlock::computeOverflow):
* Source/WebCore/rendering/svg/RenderSVGBlock.h:

Canonical link: <a href="https://commits.webkit.org/301714@main">https://commits.webkit.org/301714@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d8579855c9845bdf33c40c7c46c40920a17da333

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126743 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46385 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37352 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133700 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78377 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/63cd30ea-b9d7-4aba-a95d-e05a4b38651c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128614 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47014 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54922 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96444 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/48d6cf3b-a3b3-4c35-bf8a-766a91e482c6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129691 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37616 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113359 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76968 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e4394fc1-35f7-4510-a8c2-04ecc5710e9b) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36499 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31543 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77093 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107435 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31842 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136277 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53421 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41103 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104957 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53915 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109719 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104660 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26705 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50155 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28494 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50874 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53349 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59147 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52608 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55946 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54357 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
`No new tests (OOPS!).` will probably trip up merge-queue

fantasai commented on Source/WebCore/rendering/RenderBox.h:
```diff
@@ -165,11 +165,18 @@ class RenderBox : public RenderBoxModelObject {
+        RecomputeFloats = 1 << 0,
+    };
+    virtual void addOverflowFromInFlowChildren(OptionSet<ComputeOverflowOptions> = { });
+    void addOverflowFromChild(const RenderBox& child, OptionSet<ComputeOverflowOptions> = { });
```
Alan points out these are not always children, so need to rename this to something else that's more correct.

fantasai replied:
Renamed to addOverflowFromContainedBox

fantasai committed:
Clean up RenderBox overflow methods
https://bugs.webkit.org/show_bug.cgi?id=300835
rdar://162722242

Reviewed by Alan Baradlay.

The various computeOverflow/addOverflow methods are kinda tangled up. Clean it up.
- Shifts method declarations to the correct superclass/subclass level
  (adjusting names as necessary), and in the right section of the .h file.
- Simplifies parameters and sets us up for additional options in the future.
- Encapsulates memory management of m_overflow.

* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::computeOverflow):
(WebCore::RenderBlock::addOverflowFromOutOfFlowBoxes):
(WebCore::RenderBlock::addVisualOverflowFromTheme):
(WebCore::RenderBlock::simplifiedLayout):
(WebCore::RenderBlock::addOverflowFromChildren): Deleted.
(WebCore::RenderBlock::addOverflowFromBlockChildren): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
(WebCore::RenderBlock::computeOverflow):
(WebCore::RenderBlock::addOverflowFromInlineChildren): Deleted.
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::computeOverflow):
(WebCore::RenderBlockFlow::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderBlockFlow.h:
(WebCore::RenderBlockFlow::computeOverflow):
(WebCore::RenderBlockFlow::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::addOverflowFromInFlowChildren):
(WebCore::RenderBox::addOverflowFromContainedBox):
(WebCore::RenderBox::addOverflowWithRendererOffset):
(WebCore::RenderBox::addVisualOverflow):
(WebCore::RenderBox::percentageLogicalHeightIsResolvable const):
(WebCore::RenderBox::hasUnsplittableScrollingOverflow const):
(WebCore::RenderBox::isUnsplittableForPagination const):
(WebCore::RenderBox::addOverflowFromInFlowChildOrAbsolutePositionedDescendant): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::contentOverflowRect const):
(WebCore::RenderGrid::computeOverflow): Deleted.
* Source/WebCore/rendering/RenderGrid.h:
* Source/WebCore/rendering/RenderMultiColumnSet.cpp:
(WebCore::RenderMultiColumnSet::addOverflowFromInFlowChildren):
(WebCore::RenderMultiColumnSet::addOverflowFromChildren): Deleted.
* Source/WebCore/rendering/RenderMultiColumnSet.h:
* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::addOverflowFromInFlowChildren):
(WebCore::RenderTable::addOverflowFromChildren): Deleted.
* Source/WebCore/rendering/RenderTable.h:
(WebCore::RenderTable::addOverflowFromInFlowChildren):
* Source/WebCore/rendering/RenderTableSection.cpp:
(WebCore::RenderTableSection::computeOverflowFromCells):
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
(WebCore::RenderSVGBlock::computeOverflow):
* Source/WebCore/rendering/svg/RenderSVGBlock.h:

Canonical link: https://commits.webkit.org/301714@main
